### ISSUE TITLE
torch.nn.utils.clip_grad applies the power twice?

### ISSUE THREAD

**Issue by jiamenguk** [pytorch/pytorch#6047]
I was using the [clip_grad in torch.nn.utils](https://github.com/pytorch/pytorch/blob/master/torch/nn/utils/clip_grad.py), which normalises the gradients for each of the parameters by the sum of a power norm of the gradient, however the code itself looks something like this:

```total_norm = 0
for p in parameters:
    param_norm = p.grad.data.norm(norm_type)
    total_norm += param_norm ** norm_type
total_norm = total_norm ** (1. / norm_type)
```

Where `norm_type` is the power of the norm, such as 2 for L2-norms. Maybe I've had too much coffee today, but that looks like the norm is applied twice before being added to the total norm?

Have a pull request ready if this wasn't intended.

**Comment by jiamenguk** [comment]
Sorry, was being really stupid
